pkg/controller/daemonsets.go

```diff
@@ -66,10 +66,10 @@ func (c *StashController) runDaemonSetInjector(key string) error {
 	} else {
 		glog.Infof("Sync/Add/Update for DaemonSet %s\n", key)
 
-		ds := obj.(*extensions.DaemonSet)
+		ds := obj.(*extensions.DaemonSet).DeepCopy()
 		ds.GetObjectKind().SetGroupVersionKind(extensions.SchemeGroupVersion.WithKind(api.KindDaemonSet))
 
-		w, err := workload.ConvertToWorkload(ds.DeepCopy())
+		w, err := workload.ConvertToWorkload(ds)
```

Comment:
I had done that. But it has a problem. ds is a pointer. Hence, when we mutate w , ds also get changed. So patch method don't patch daemonset as ds and w are same.

Reply:
Good point. Should be fixed now. But I also had to add https://github.com/appscode/kutil/pull/107 to avoid an extra DeepCopy()